### PR TITLE
Remove mention of cluster virtualization / multitenancy from v23.1 release notes

### PR DESCRIPTION
**Pull request by mdlinville**
Remove mention of cluster virtualization / multitenancy from v23.1 release notes

Case-insensitive search strings:
- `*virtual cluster*`
- `*cluster virtual*`
- `tenant` (some occurrences were left in place)
- `*multiten*`
- `*multi-ten*`

Also remove the note for https://github.com/cockroachdb/cockroach/pull/92357

### Previews
[releases/v23.1.md](https://deploy-preview-17982--cockroachdb-docs.netlify.app/docs/releases/v23.1.html)

### CODE REVIEW

File: src/current/_includes/releases/v23.1/v23.1.0-alpha.1.md [hunk 1]
```diff
@@ -180,7 +180,6 @@ Release Date: December 19, 2022
 - The **Transaction resource usage** card on the **Transaction Fingerprin**t page now includes an "Idle latency" row, representing the time spent by the application performing other work while holding this transaction open. [#92951][#92951]
 - The **Databases table** page now displays all the grants in a single row per user. [#92871][#92871]
 - Added a goroutine scheduling latency graph to the **Overload** dashboard in the DB Console. It shows what the per-node p99 scheduling latency is for goroutines. [#93217][#93217]
-- Added a top-level dropdown in the DB Console which lists tenants the user has logged into. If the cluster is not a multi-tenant host, the dropdown is not displayed. [#92694][#92694]
 - CockroachDB now prevents polling `/settings`, `/nodes_ui,` and `/cluster` endpoints on incorrect login. [#93211][#93211]
 - The **Statement** and **Transaction** pages for tenant clusters gained region columns and filters for multi-region tenants. [#92357][#92357]
```
Comment (taroface) [inline src/current/_includes/releases/v23.1/v23.1.0-alpha.1.md:184]: Should this one also be removed?

**Review by taroface — APPROVED** [review]
LGTM. One question.

File: src/current/_includes/releases/v23.1/v23.1.0-alpha.1.md (hunk 1 above)
Comment (mdlinville) [reply src/current/_includes/releases/v23.1/v23.1.0-alpha.1.md:184]: Yeah let me do a pass for "tenant" too. Good catch.

File: src/current/_includes/releases/v23.1/v23.1.0-alpha.1.md [hunk 2]
```diff
@@ -112,10 +110,8 @@ Release Date: December 19, 2022
...
 - CockroachDB now uses response data rather than just the request span in the [load-based splitter](https://www.cockroachlabs.com/docs/v23.1/load-based-splitting) to pass more accurate data about the keys iterated over to the load splitter to find a suitable split key, enabling the load splitter to find a split key under heavy range query workloads. [#89217][#89217]
 - Added the `replicas.leaders_invalid_lease` metric, which indicates how many replicas are Raft group leaders but holding invalid leases. [#91179][#91179]
-- SQL tenants now support the [HTTP endpoint](https://www.cockroachlabs.com/docs/v23.1/cluster-api) under `/api/v2/sql` which allows the caller to execute an HTTP request containing SQL statements to execute. The JSON response contains the results. This endpoints works identically as on a non-tenant server, except that it naturally scopes to the target tenant for SQL execution. [#91323][#91323]
```
Comment (mdlinville) [inline src/current/_includes/releases/v23.1/v23.1.0-alpha.1.md:115]: @dikshant Please review only this removal. :)

Comment (dikshant) [reply src/current/_includes/releases/v23.1/v23.1.0-alpha.1.md:115]: LGTM!

**Review by dikshant — APPROVED** [review]

**Pull request merged**